app/api/discount_codes.py

```diff
@@ -23,14 +23,27 @@ class Meta:
         self_view_kwargs = {'id': '<id>'}
         inflect = dasherize
 
-    @validates_schema
-    def validate_quantity(self, data):
-        if 'max_quantity' in data and 'min_quantity' in data:
-            if data['max_quantity'] < data['min_quantity']:
+    @validates_schema(pass_original=True)
+    def validate_quantity(self, data, original_data):
+        original_data_attributes = original_data['data']['attributes']
+        discount_code = DiscountCode.query.filter_by(id=original_data['data']['id']).one()
+        if 'min_quantity' in original_data_attributes:
```

Comment:
@poush 
As I mentioned above.
> It is the validation of partial dataset .
It is for the case if the user provided only the `max_quantity` ( or min-quantity ) and if it's less than `min-quantity` ( or max-quantity ) which is already stored in the database.

> it makes sense to check on validated data object i.e, data here.
data here is already processed with initial validations.

What **initial validations** are you talking about on **partial datasets** ( i.e. ones where only either of the parameters is given ) ?

Yes I do agree that **when both the parameters are present then we can use data attribute**.
Like in here 
```
if 'max_quantity' in original_data_attributes and 'min_quantity' in original_data_attributes:
```
and
```
if 'tickets_number' in original_data_attributes and 'min_quantity' in original_data_attributes:
```
But I kept it just for uniformity .

As far as these are concerned 
```
min_quantity = fields.Integer(validate=lambda n: n >= 0)
max_quantity = fields.Integer(validate=lambda n: n >= 0)
```
These will be checked anyhow.
Correct me if I misunderstood your question.